test/validation/api/traffic_mngr/traffic_mngr.c

```diff
@@ -1951,7 +1991,12 @@ static int destroy_all_shaper_profiles(void)
 	uint32_t        idx;
 	int             rc;
 
-	for (idx = 0; idx < NUM_SHAPER_PROFILES; idx++) {
+	if (num_shaper_profiles >= MAX_SHAPER_PROFILES) {
```

Comment:
These checks that have been added in various destroy functions look a bit out-of-place. The destroy function could still destroy the objects it can, so this check does not seem to be needed for the destroy function. This looks more like some internal consistency checking that would be better done at creation/allocation time than way later here in the destroy functions.